DataStructures/HashTables/HashNode.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace DataStructures.HashTables
+{
+    public class HashNode<TValue>
+    {
+        public HashNode(int bucket, int key, TValue value)
+        {
+            Bucket = bucket;
+            Key = key;
+            Value = value;
+        }
+        public int Bucket { get; set; }
+        public int Key { get; set; }
```

Comment:
Looks like this might be the hash, not the actual key? Storing the hash can be a nice optimization, because comparing `key`s can get expensive.